lib/vmpooler/providers/vsphere.rb

```diff
@@ -839,7 +839,9 @@ def find_least_used_host(cluster, connection, datacentername)
         def find_cluster(cluster, connection, datacentername)
           datacenter = connection.serviceInstance.find_datacenter(datacentername)
           raise("Datacenter #{datacentername} does not exist") if datacenter.nil?
-          datacenter.hostFolder.children.find { |cluster_object| cluster_object.name == cluster }
+          found_cluster = datacenter.hostFolder.traverse(cluster)
```

Comment:
this requires changes to `cluster` from a named string to a `path` (see equivalent method in rbvmomi https://www.rubydoc.info/github/vmware/rbvmomi/RbVmomi%2FVIM%2FDatacenter:find_compute_resource). So other code changes up the stack would be needed.

My understanding is that it would return the first found item, not necessarily the right item, for example if you have multiple clusters in the datacenter. This is represented in the #find_cluster spec tests, so the PR here should not break those tests (or require changes in line with the logic change) 

https://github.com/puppetlabs/vmpooler/blob/98a84ab239c204da8022a87e0aa66307dd14f718/spec/unit/providers/vsphere_spec.rb#L2883

That being said I agree with the original intention of the PR to fix finding folders that have the same name as the target cluster, but I believe the solution should be with checking the type of the object found until it is a `RbVmomi::VIM::ComputeResource`